### PR TITLE
[mono][jit] Enable all X86Base intrinsics

### PR DESCRIPTION
**Pull request by jandupej**
Support for `System.Runtime.Intrinsics.X86.DivRem` is added here. This should mirror CoreCLR's https://github.com/dotnet/runtime/pull/66551 and address https://github.com/dotnet/runtime/issues/82195.

EDIT: This also adds support for `DivRem` on LLVM, as well as `BitScanForward` and `BitScanReverse` on mini.

### CODE REVIEW

File: src/mono/mono/mini/mini-ops.h [hunk 1]
```diff
@@ -1184,6 +1184,15 @@ MINI_OP3(OP_MULX_HL64, "mulxhl64", LREG, LREG, LREG, LREG)
 
 #endif
 
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
+MINI_OP3(OP_X86_LDIVREM, "long_divrem", LREG, LREG, LREG, LREG)
```
Comment (vargaz) [inline src/mono/mono/mini/mini-ops.h:1188]: These could use some comments as to what they exactly do and how the 2 opcodes work together.

**Comment by jandupej** [comment]
All CI failures are known issues unrelated to this PR.

**Review by fanyang-mono — APPROVED** [review]
LGTM!

**Comment by jandupej** [comment]
Got an occurrence of https://github.com/dotnet/runtime/issues/90458

**Pull request merged**